src/db.c

```diff
@@ -2049,23 +2125,25 @@ int expireIfNeeded(redisDb *db, robj *key, int flags) {
 
 /* Increases size of the main db to match desired number. In cluster mode resizes all individual dictionaries for slots that
  * this node owns. */
-int expandDb(const redisDb *db, uint64_t db_size, unsigned dictType) {
-    serverAssert(dictType < DICT_TYPE_MAX);
+int expandDb(const redisDb *db, uint64_t db_size, dbKeyType keyType) {
```

Comment:
i'll state again that i think this method should get an argument weather to use dictExpand or dictTryExpand.
in case it was called from rdb.c, i think it should panic in zmalloc and not fail gracefully.

Reply:
Sure.